clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -334,23 +338,27 @@ public boolean isInternal() {
 
     }
 
+    // TODO: rename this class to MetadataResponsePartitionState
```

Comment:
Since it's an inner class of `MetadataResponse`, there is no need to also add `MetadataResponse` as a prefix to the inner class name.